README.md

```diff
@@ -13,128 +13,339 @@
 | Python Linux       | ![master python linux e2e](https://github.com/Azure/functions-action/workflows/RUN_E2E_TESTS_PYTHON37_LCON/badge.svg?branch=master) | ![dev python linux e2e](https://github.com/Azure/functions-action/workflows/RUN_E2E_TESTS_PYTHON37_LCON/badge.svg?branch=dev) |
 
+The Azure Functions action is used in a [GitHub Actions workflow](https://docs.github.com/en/actions/about-github-actions/understanding-github-actions) to deploy packaged project code to an existing [function app](https://azure.microsoft.com/services/functions/) hosted in Azure. Using this action, you can create continuous workflow automation that builds, authenticates, and deploys code to your function app when you make changes in the GitHub repository.
 
-With the Azure Functions GitHub Action, you can automate your workflow to deploy [Azure Functions](https://azure.microsoft.com/en-us/services/functions/).
+The Azure Functions action is defined in this [action.yml](https://github.com/Azure/functions-action/blob/master/action.yml) file. To learn about specific parameters, see the [Reference section](#parameter-reference) in this readme.
 
-Get started today with a [free Azure account](https://azure.com/free/open-source)!
+## Getting started
 
-The repository contains a GitHub Action to deploy your function app project into Azure Functions. If you are looking for a GitHub Action to deploy your customized container image into an Azure Functions container, please consider using [functions-container-action](https://github.com/Azure/functions-container-action).
+You can create a workflow for your function app deployments in one of these ways:
 
-The definition of this GitHub Action is in [action.yml](https://github.com/Azure/functions-action/blob/master/action.yml).
++ [**Workflow templates**](#workflow-templates): choose a workflow template that matches your function app's code language, operating system, and hosting plan. Add this workflow (.yaml) file to the `.github/workflows/` folder of the repository that contains your project code.
 
-[Kudu zip deploy](https://github.com/projectkudu/kudu/wiki/Deploying-from-a-zip-file-or-url) method is used by the action for deployment of Functions.
++ [**Azure portal**](https://portal.azure.com): you can create a workflow (.yaml) file directly in your repository from your app's **Deployment center** page in the portal by following the detailed instructions in [Continuous delivery by using GitHub Actions](https://learn.microsoft.com/azure/azure-functions/functions-how-to-github-actions?pivots=method-portal).
 
-# End-to-End Sample workflow
+>[!IMPORTANT]
+>If you're looking for a GitHub Action to deploy your customized container image to Azure Functions, instead use [functions-container-action](https://github.com/Azure/functions-container-action).
```

Comment:
Should this be moved up? It seems like this should precede "getting started"

Reply:
I'm ok with moving it up. It was in this position before, so I left it. I think it makes sense to precede "getting started" with this call out.

Rocket emoji so I know to come back to this.